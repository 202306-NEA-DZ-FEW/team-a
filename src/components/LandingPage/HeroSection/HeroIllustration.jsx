function HeroIllustration() {
  return <div>HeroIllustration</div>;
}

export default HeroIllustration;
