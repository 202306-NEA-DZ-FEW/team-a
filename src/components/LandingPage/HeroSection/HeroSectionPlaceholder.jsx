import Container from "@/components/container";

function HeroSectionPlaceholder() {
  return (
    <Container>
      <div className='flex flex-col justify-between items-center lg:flex-row-reverse gap-10 lg:gap-4 lg:min-h-screen my-20 lg:my-0'>
        <div
          id='illustration'
          className='md:w-1/2 w-full h-96 rounded-xl md:px-0 bg-gray-300 animate-pulse'
        ></div>
        <div className='flex flex-col justify-start lg:w-1/2 gap-2'>
          <div className='w-96  h-6 bg-gray-300 animate-pulse rounded-full mb-8'></div>
          <div className='w-1/3 h-3 bg-gray-300 animate-pulse rounded-full'></div>
          <div className='w-1/2 h-3 bg-gray-300 animate-pulse rounded-full'></div>
          <div className='w-1/4 h-3 bg-gray-300 animate-pulse rounded-full'></div>
          <div className='w-1/5 h-3 bg-gray-300 animate-pulse rounded-full'></div>
          <div className='w-1/2 h-3 bg-gray-300 animate-pulse rounded-full'></div>
          <div className='w-1/4 h-10 bg-gray-300 animate-pulse rounded-xl mt-4'></div>
        </div>
      </div>
    </Container>
  );
}

export default HeroSectionPlaceholder;
