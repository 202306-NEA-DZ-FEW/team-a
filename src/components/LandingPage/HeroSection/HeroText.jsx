function HeroText({ text }) {
  return <div>{text}</div>;
}

export default HeroText;
