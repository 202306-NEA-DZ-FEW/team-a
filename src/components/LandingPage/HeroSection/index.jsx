import { motion } from "framer-motion";
import Image from "next/image";
import { useTranslation } from "next-i18next";
import { BiDonateHeart } from "react-icons/bi";

import { Triangles, Xhorizontal, Xvertical } from "@/components/icons";
function HeroSection() {
  const { t } = useTranslation();
  return (
    <>
      <div className='flex flex-col items-center lg:flex-row-reverse lg:min-h-screen'>
        <figure className='relative w-full lg:w-[55%] h-1/2 lg:h-screen'>
          <Image
            src='/images/hero-image.png'
            alt='hero image'
            width={4928}
            height={3264}
            placeholder='blur'
            blurDataURL='blur.jpg'
            className='w-full h-full object-cover overflow-auto'
            priority
          />

          <Triangles className='absolute top-[15%] left-[-10%] fill-current w-[16%] lg:block hidden opacity-75 z-0' />
          <Xvertical className='absolute bottom-[5%] right-[3%] w-[2%] lg:block hidden z-0' />
        </figure>
        <div
          id='text'
          className='flex flex-col py-8 justify-center w-full lg:w-[45%] lg:gap-6 gap-4 px-6 lg:h-screen z-10'
        >
          <motion.h2
            className='text-4xl lg:text-5xl font-black capitalize'
            initial={{ y: 30, opacity: 0 }}
            animate={{ y: 0, opacity: 1 }}
            transition={{ duration: 0.4 }}
            viewport={{ once: true }}
          >
            {/* {t("landingPage:hero:title")} */}
            The power of collective
            <span className='text-accent'> Giving.</span>
          </motion.h2>
          <motion.p
            className='text-start font-light text-md md:text-xl'
            initial={{ y: 30, opacity: 0 }}
            animate={{ y: 0, opacity: 1 }}
            transition={{ duration: 0.4, delay: 0.5 }}
          >
            You offer more than possessions when you give, true giving is an
            extension of yourself
          </motion.p>
          <motion.button
            className='mb-10 btn lg:self-start lg:min-w-fit text-current bg-opacity-50  rounded-full flex items-center btn-secondary'
            initial={{ y: 30, opacity: 0 }}
            animate={{ y: 0, opacity: 1 }}
            transition={{ duration: 0.4, delay: 0.7 }}
          >
            {t("common:buttons:donate")}
            <span className='text-lg'>
              <BiDonateHeart />
            </span>
          </motion.button>
          <motion.div
            className='relative w-full rounded-2xl bg-opacity-50 ring-4 ring-white ring-opacity-20 bg-secondary flex lg:flex-row lg:justify-between flex-col items-center py-8 gap-4 px-4 z-10'
            initial={{ y: 30, opacity: 0 }}
            animate={{ y: 0, opacity: 1 }}
            transition={{ duration: 0.4, delay: 1 }}
          >
            <p className='flex items-center  gap-2 text-xs font-light'>
              <span className='text-lg font-black'>30k+</span> Ramadan Meal
              Outreach
            </p>
            <p className='flex items-center  gap-2 text-xs font-light'>
              <span className='text-lg font-black'>200+</span> In Orphan
              Education Support
            </p>
            <p className='flex items-center gap-2 text-xs font-light'>
              <span className='text-lg font-black'>25k+</span> In Refugee Aid
              Program
            </p>
            <Xhorizontal className='absolute -top-2 left-7 fill-current w-36' />
          </motion.div>
        </div>
      </div>
    </>
  );
}

export default HeroSection;
