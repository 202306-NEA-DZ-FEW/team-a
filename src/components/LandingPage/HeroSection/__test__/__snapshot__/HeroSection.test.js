import renderer from "react-test-renderer";

import HeroSection from "../..";

it("renders correctly", () => {
  const tree = renderer.create(<HeroSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
