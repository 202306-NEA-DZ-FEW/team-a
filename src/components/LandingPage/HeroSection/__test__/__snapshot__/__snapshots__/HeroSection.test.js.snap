// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-w-6xl mx-auto undefined"
>
  <div
    className="flex flex-col justify-center items-center md:flex-row-reverse gap-4  min-h-screen"
  >
    <div
      className="md:w-1/2 p-6 m-2"
      id="illustration"
    >
      <img
        alt="hero illustration"
        data-nimg="1"
        decoding="async"
        height={40}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={40}
      />
    </div>
    <div
      className="flex flex-col justify-center items-center  p-2 md:w-1/2 gap-2"
      id="text"
    >
      <h1
        className="text-5xl font-futuraBlack"
      >
        let's share
      </h1>
      <p
        className="text-start p-8"
      >
        Welcome to Let's Share! Where everything you may not need is someone else's necessity. From clothing to furniture, we are here to be the link between you and people in need. Together, we can make a significant difference in the lives of those who require a helping hand. Join us in spreading warmth, kindness, and hope. Let's create a community bound by the spirit of giving and compassion, one item at a time.
      </p>
      <button
        className="text-base btn btn-primary py-2 px-4 text-white font-bold hover:bg-green-900"
      >
        Donate now
      </button>
    </div>
  </div>
</div>
`;
